service/service.go

```diff
@@ -0,0 +1,44 @@
+package service
+
+import (
+	"crypto-price-tracker/config"
+	"crypto-price-tracker/provider"
+	"errors"
+	"github.com/gin-gonic/gin"
+	"time"
+)
+
+func CheckAPIExpiry() bool {
+	if time.Since(config.LastFetchedAPITime) > config.CacheExpiry {
```

Comment:
the value `config.LastFetchedAPITime` is being used before it is called.
Which should not be the case, else it will take 0 value of time `// The zero value of type Time is January 1, year 1, 00:00:00.000000000 UTC.`
And make the further execution with that.

We can have a logic that if the time is not set, or if the  `ime.Since(config.LastFetchedAPITime) > config.CacheExpiry` then call the API.

Reply:
When its set to the 0 value initially the time.Since will return a very large value which will be > than config.CacheExpiry, So I kept it like this only although it better to keep a check as you suggested so updated it with this, if config.LastFetchedAPITime.IsZero() || time.Since(config.LastFetchedAPITime) > config.CacheExpiry